rasa/shared/core/domain.py

```diff
@@ -362,7 +365,7 @@ def merge_lists(list1: List[Any], list2: List[Any]) -> List[Any]:
 
     @staticmethod
     def merge_lists_of_dicts(
-        self,
+        self: Type[D],
```

Comment:
As a `staticmethod`- there shouldn't be any `self`or `cls`, right?

Reply:
That's what I thought, but as this method calls `merge_dicts` it needs to call`self.merge_dicts`. And then the mypy type check failed with an `untyped-defs` error.

Reply:
In that case `merge_dicts`should be declared as `@staticmethod` too, because as far as I can see it doesn't need a `self` instance. 

`mypy` complains as `self` here does not refer to the Domain instance but is a independent parameter that has to be passed to `merge_list_of_dicts` as first parameter :-(